### DISCUSSION TITLE
[Bug]: image targets 'amd64', cannot run on 'arm64'

### DISCUSSION THREAD

**Discussion by scott-love** [neurodesk/neurodesk.github.io-hugo#436] [Bug Report]
### Version

1.6

### What platform are you seeing the problem on?

Local PC

### What happened

I have a nipype pipeline that was working a few months ago. Now on the same computer (actually 2) it doesn't work. 

The nipype workflow gets created but the first time it tries to use an ANTs function (ImageMath) it crashes. I tried running ImageMath in a terminal and I get this error: FATAL:   image targets 'amd64', cannot run on 'arm64'

I tried running an AFNI function in the terminal and get the same error.

I am working on Apple M1 Max (or M3 Pro)
OS 14.6.1

I tried turning on and off Rosetta, which made no difference. 
I have not tried Podman.

Can anyone see a solution to this?

### How to reproduce

In a Neurodesk app terminal do:
```
ml ants/2.3.5
ImageMath

```

### Relevant log output or screenshot

```shell
(base) jovyan@neurodesktop-2024-05-25:~$ ml ants/2.3.5
(base) jovyan@neurodesktop-2024-05-25:~$ ImageMath 
FATAL:   image targets 'amd64', cannot run on 'arm64'
```

**Comment by stebo85** [comment]
Dear @scott-love,

I can reproduce this problem on the latest MacOS (14.6.1) and latest Docker version (4.33.0 (160616)) - I am assuming this is a Docker bug for now, because it works in Podman:

<img width="1121" alt="image" src="https://github.com/user-attachments/assets/853c6a2a-6665-4011-81af-fe5cf51dbc3a">

We are also planning to get native arm64 containers soon - so hopefully the podman workaround works for you until docker pushed an update or until we have native arm64 containers.

**Reply by scott-love** [reply]
Thanks for the info @stebo85.
I installed Podman and after a few hurdles managed to get the pipeline running. It has reached all processes yet but at least it got started :)
I did find one quirk that I will open another bug report about.

**Reply by stebo85** [reply]
great - thank you :)

**Comment by stebo85** [comment]
Dear @scott-love, we have now released a new version of neurodeskapp 1.6.1 which has a workaround for this docker bug.

**Reply by scott-love** [reply]
Dear @stebo85, 
Until today, I had not checked if this was working for me. However, today using neurodeskapp 1.7.0 I tried to switch back to docker 4.36.0 (175267) and unfortunately got exactly the same error as above. I tried both with and without Rosetta. 

I send this simply for information because fortunately using Podman still works for me.
Thanks,
Scott

**Reply by stebo85** [reply]
@scott-love - thank you for raising that. We have a suspicion what this could be. To confirm the suspicion can you let us know which applications are affected for you? It shouldn't be all - just a few ones? If that's the case we can fix it hopefully quickly.

**Reply by scott-love** [reply]
Based on what I use In my pipeline:
I tested in a neurodesk terminal ants/2.3.5 and convert3d/1.0.0 and both gave the same error as above
I also tested afni/23.3.02 and fsl/6.0.7.8 in the terminal, which did **not** give the error above.

Inside neurodeskapp virtual desktop I tried to load SPM12GUI and EEGLAB2020 GUI and both crash straightaway.

Hope this helps @stebo85, let me know if you want me to test anymore.

**Reply by stebo85** [reply]
that helps - we broke this partially in the last update and will fix it with the next release again.

**Reply by stebo85** [reply]
Dear @scott-love,
We released a new version of Neurodesk today and all these issues are fixed now (it was a combination of problems caused by MacOS Rosetta 2.0 and Docker). Make sure to run Neurodeskapp version 1.8.0 with Neurodesktop image version 2025-01-29 in Docker Desktop 4.37.2 (179585) with latest MacOS updates 15.3 (24D60) and run the RDP desktop interface:

<img width="1493" alt="image" src="https://github.com/user-attachments/assets/e634a308-08cd-441d-9261-7e358376f62e" />

**Reply by scott-love** [reply]
@stebo85 - Excellant, I can confirm that this is now working using Docker (both with and without Rosetta, which seems faster) and Podman.
